app/controllers/school_search_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SchoolSearchController < ApplicationController
+  def create
+    search_schools
+    render status: errors.blank? ? :ok : :bad_request
+  end
+
+  private
+
+  def search_schools
+    if params[:query].blank?
+      errors.push("Expected required parameter 'query' to be set")
+      return
+    end
+
+    begin
+      @schools = School.search(params[:query])
+    rescue ArgumentError => e
+      raise unless e.message == "search_term must have a minimum of 4 characters"
```

Comment:
Should we extract this error message to a constant on `School`? Feels a bit brittle having it hard coded in multiple places.

Reply:
The error message is actually different in different contexts. When interacting with the JSON enpoint, we refer to the parameter (though I now notice that it's still `search_term` when it should be `query`).

Reply:
Ah nevermind. You're right. Got my wires crossed.